### DISCUSSION TITLE
go lint is giving errors while executing a new rule implementation

### DISCUSSION THREAD

**Discussion by nagarajtrivedi** [hyperjumptech/grule-rule-engine#240] [Q&A] [answered]
Hi all, I am executing a rule implementation file through command make test-short
But it is throwing below error. Let me know what could be the problem.
**golint -set_exit_status builder/... engine/... examples/... ast/... pkg/... antlr/. model/...
examples/CheckMultipleTimesRulesExecution.go:27:6: exported type MyData should have comment or be unexported
examples/CheckMultipleTimesRulesExecution.go:32:1: exported method MyData.GetMsg should have comment or be unexported
examples/CheckMultipleTimesRulesExecution.go:36:1: exported function TestMultipleTimesRuleExec should have comment or be unexported
Makefile:13: recipe for target 'lint' failed**

**Comment by newm4n** [comment, marked as answer]
Its clearly stated in the files, and line numbers.
You need to make comments explaining the type, method and functions. Such as

```go
// MyData is a struct about this and that
type MyData struct {
     ...
}

// GetMsg will do this and that
func (md *MyData) GetMsg() {
      ...
}

// TestMultipleTimesRuleExec is to test this and that ...
func TestMultipleTimesRuleExec(t *testing.T) {
    ...
}
```

**Reply by nagarajtrivedi** [reply]
Yes, I got it. I tried whatever suggested by the lint and it worked.
